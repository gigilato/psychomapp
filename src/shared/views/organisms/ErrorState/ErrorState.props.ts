import { IBoxProps } from '$atoms'

export interface IErrorStateProps extends IBoxProps {
  type?: 'error' | 'empty'
  message?: string
  width?: number
  height?: number
}
