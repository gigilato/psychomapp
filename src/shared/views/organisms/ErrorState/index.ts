export * from './ErrorState'
export * from './ErrorState.props'
