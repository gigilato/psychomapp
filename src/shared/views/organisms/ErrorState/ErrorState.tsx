import clsx from 'clsx'

import { Center, Image, Text } from '$atoms'
import { i18n } from '$infra/i18n'
import { IErrorStateProps } from '$organisms/ErrorState/ErrorState.props'

export const ErrorState = ({
  type = 'error',
  message,
  className,
  width,
  height,
  ...props
}: IErrorStateProps) => {
  return (
    <Center className={clsx('self-center', className)} {...props}>
      <Image source={type} width={width ?? 200} height={height} />
      <Text className="mt-l">
        {message ?? i18n.t(type === 'empty' ? 'errors.empty' : 'errors.default')}
      </Text>
    </Center>
  )
}
