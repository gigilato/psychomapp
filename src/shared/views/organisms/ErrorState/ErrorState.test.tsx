import { render } from '@testing-library/react-native'
import React from 'react'

import { ErrorState } from './ErrorState'

describe('ErrorState', () => {
  it('should render correctly', () => {
    const { toJSON } = render(<ErrorState type="empty" />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should render with custom message', () => {
    const { toJSON } = render(<ErrorState type="empty" message="custom message" />)
    expect(toJSON()).toMatchSnapshot()
  })
})
