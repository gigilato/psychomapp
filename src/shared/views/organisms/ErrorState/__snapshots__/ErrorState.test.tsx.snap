// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorState should render correctly 1`] = `
<View
  style={
    [
      {
        "justifyContent": "center",
      },
      {
        "alignItems": "center",
      },
      {
        "alignSelf": "center",
      },
      [
        undefined,
        undefined,
      ],
    ]
  }
>
  <ViewManagerAdapter_ExpoImage
    contentFit="contain"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={100}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    source={[]}
    style={
      {
        "height": 100,
        "width": 100,
      }
    }
    transition={null}
    width={100}
  />
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Regular",
        },
        {
          "fontSize": 16,
        },
        {
          "lineHeight": 18,
        },
        {
          "marginTop": 24,
        },
      ]
    }
  >
    Aucun r√©sultat
  </Text>
</View>
`;

exports[`ErrorState should render with custom message 1`] = `
<View
  style={
    [
      {
        "justifyContent": "center",
      },
      {
        "alignItems": "center",
      },
      {
        "alignSelf": "center",
      },
      [
        undefined,
        undefined,
      ],
    ]
  }
>
  <ViewManagerAdapter_ExpoImage
    contentFit="contain"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={100}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    placeholder={[]}
    source={[]}
    style={
      {
        "height": 100,
        "width": 100,
      }
    }
    transition={null}
    width={100}
  />
  <Text
    style={
      [
        {
          "fontFamily": "SpaceGrotesk-Regular",
        },
        {
          "fontSize": 16,
        },
        {
          "lineHeight": 18,
        },
        {
          "marginTop": 24,
        },
      ]
    }
  >
    custom message
  </Text>
</View>
`;
